customer-satisfaction/steps/evaluation.py

```diff
@@ -1,38 +1,59 @@
 import logging
+from typing import Tuple, Annotated
 
 import mlflow
 import numpy as np
 import pandas as pd
-from model.evaluation import Evaluation
+from model.evaluator import Evaluator
 from sklearn.base import RegressorMixin
 from zenml.client import Client
-from zenml.steps import Output, step
+from zenml import step, get_step_context, log_artifact_metadata
 
 experiment_tracker = Client().active_stack.experiment_tracker
 
 
 @step(experiment_tracker=experiment_tracker.name)
 def evaluation(
-    model: RegressorMixin, x_test: pd.DataFrame, y_test: pd.Series
-) -> Output(r2_score=float, rmse=float):
-    """
+    model: RegressorMixin,
+    x_test: pd.DataFrame,
+    y_test: pd.Series
+) -> Tuple[
+    Annotated[float, "r2_score"],
+    Annotated[float, "rmse"]
+]:
+    """Evaluates the Model on the Test Dataset and returns the metrics.
+
     Args:
         model: RegressorMixin
         x_test: pd.DataFrame
         y_test: pd.Series
+
     Returns:
         r2_score: float
         rmse: float
     """
     try:
         prediction = model.predict(x_test)
-        evaluation = Evaluation()
-        r2_score = evaluation.r2_score(y_test, prediction)
+        evaluator = Evaluator()
+        r2_score = evaluator.r2_score(y_test, prediction)
+        mse = evaluator.mean_squared_error(y_test, prediction)
+        rmse = np.sqrt(mse)
+
+        # Log to MLFlow
         mlflow.log_metric("r2_score", r2_score)
-        mse = evaluation.mean_squared_error(y_test, prediction)
         mlflow.log_metric("mse", mse)
-        rmse = np.sqrt(mse)
         mlflow.log_metric("rmse", rmse)
+
+        # Also add these metrics to the Model within the ZenML Model Control Plane
+        artifact = get_step_context().model_version.get_artifact("model")
+
+        log_artifact_metadata(
+            metadata={"r2_score": float(r2_score),
+                      "mse": float(mse),
+                      "rmse": float(rmse)},

```

Comment:
great one! What i like to do is net it under a `metrics` dict so it gets properly categorized in the dashboard:

```suggestion
            metadata={"metrics": {"r2_score": float(r2_score),
                      "mse": float(mse),
                      "rmse": float(rmse)}},

```